Mlem/App/Views/Shared/Avatar/AvatarBannerView.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  AvatarBannerView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 09/05/2024.
+//
+
+import MlemMiddleware
+import NukeUI
+import SwiftUI
+
+struct AvatarBannerView: View {
+    var model: (any ProfileProviding)?
+    var type: AvatarType
+    var showEmptyBanner: Bool = false
+    var showBanner: Bool = true
+    var showAvatar: Bool = true
+
+    init<T: ProfileProviding>(_ model: T?, showEmptyBanner: Bool = false) {
+        self.model = model
+        self.type = T.avatarType
+        self.showEmptyBanner = showEmptyBanner
+    }
+    
+    init(_ model: any ProfileProviding, showEmptyBanner: Bool = false) {
+        self.model = model
+        self.type = Swift.type(of: model).avatarType
+        self.showEmptyBanner = showEmptyBanner
+    }
+    
+    init(_ model: (any ProfileProviding)?, type: AvatarType, showEmptyBanner: Bool = false) {
+        self.model = model
+        self.type = type
+        self.showEmptyBanner = showEmptyBanner
+    }
+    
+    static let bannerHeight: CGFloat = 170
+    static let avatarOverdraw: CGFloat = 40
+    static let avatarSize: CGFloat = 108
+    static let avatarPadding: CGFloat = AppConstants.postAndCommentSpacing
+    
+    @ViewBuilder
+    func avatarView() -> some View {
+        AvatarView(url: model?.avatar, type: type)
+            .frame(height: AvatarBannerView.avatarSize)
+    }
```

Comment:
*nit*

Per new CONTRIBUTING; should also be moved below `body`

```suggestion
    @ViewBuilder
    var avatarView: some View {
        AvatarView(url: model?.avatar, type: type)
            .frame(height: AvatarBannerView.avatarSize)
    }
```